x-pack/plugin/sql/src/main/java/org/elasticsearch/xpack/sql/planner/QueryFolder.java

```diff
@@ -701,6 +701,10 @@ protected PhysicalPlan rule(OrderExec plan) {
                 EsQueryExec exec = (EsQueryExec) plan.child();
                 QueryContainer qContainer = exec.queryContainer();
 
+                if (qContainer.sort().isEmpty() == false) {
+                    throw new SqlIllegalArgumentException("QueryContainer already defines sort order");
```

Comment:
I think we should be defensive here because the naive approach to just prepend additional sort orders leads to https://github.com/elastic/elasticsearch/issues/76013 .

Alternatively, `FoldOrderBy` (and `Verifier`) could take over the responsibility to merge `OrderBy`s from `PushDownAndCombineOrderBy`.

Reply:
> Alternatively, FoldOrderBy (and Verifier) could take over the responsibility to merge OrderBys from PushDownAndCombineOrderBy.

I like this better.